src/lib/util.ts

```diff
@@ -66,3 +67,121 @@ export const formatParamValueForDisplay = (value: number | string) => {
 };
 
 export const cloneJSON = (value: JsonMap): JsonMap => JSON.parse(JSON.stringify(value));
+
+export const formatMIDIMappingToDisplay = (type: MIDIMetaMappingType, mapping: MIDIMetaMapping): string => {
+	switch (type) {
+		case MIDIMetaMappingType.ChannelPressure: {
+			return `CPRESS/${(mapping as MIDIChannelPressureMetaMapping).chanpress}`;
+		}
+		case MIDIMetaMappingType.ControlChange: {
+			return `CC#${(mapping as MIDIControlChangeMetaMapping).ctrl}/${(mapping as MIDIControlChangeMetaMapping).chan}`;
+		}
+		case MIDIMetaMappingType.KeyPressure: {
+			return `KPRESS#${(mapping as MIDIKeypressMetaMapping).keypress}/${(mapping as MIDIKeypressMetaMapping).chan}`;
+		}
+		case MIDIMetaMappingType.Note: {
+			return `NOTE#${(mapping as MIDINoteMetaMapping).note}/${(mapping as MIDINoteMetaMapping).chan}`;
+		}
+		case MIDIMetaMappingType.PitchBend: {
+			return `BEND/${(mapping as MIDIPitchBendMetaMapping).bend}`;
+		}
+		case MIDIMetaMappingType.ProgramChange: {
+			return `PRGCHG/${(mapping as MIDIProgramChangeMetaMapping).prgchg}`;
+		}
+		default: {
+			// eslint-disable-next-line @typescript-eslint/no-unused-vars
+			const _exhaustive: never = type;
+			throw new Error(`Unknown MIDIMappingType "${type}"`);
+		}
+	}
+};
+
+const midiMetaRegexp: Record<MIDIMetaMappingType, RegExp> = {
+	[MIDIMetaMappingType.ChannelPressure]: /^CPRESS\/(?<chanpress>[0-9]{1,2})$/,
+	[MIDIMetaMappingType.ControlChange]: /^CC#(?<ctrl>[0-9]{1,3})\/(?<chan>[0-9]{1,2})$/,
+	[MIDIMetaMappingType.KeyPressure]: /^KPRESS#(?<keypress>[0-9]{1,3})\/(?<chan>[0-9]{1,2})$/,
+	[MIDIMetaMappingType.Note]: /^NOTE#(?<note>[0-9]{1,3})\/(?<chan>[0-9]{1,2})$/,
+	[MIDIMetaMappingType.PitchBend]: /^BEND\/(?<bend>[0-9]{1,2})$/,
+	[MIDIMetaMappingType.ProgramChange]: /^PRGCHG\/(?<prgchg>[0-9]{1,2})$/
+};
+
+export const parseMIDIMappingDisplayValue = (value: string): false | { type: MIDIMetaMappingType, mapping: MIDIMetaMapping } => {
+	for (const [mappingType, reg] of Object.entries(midiMetaRegexp) as Array<[MIDIMetaMappingType, RegExp]>) {
+		const match = value.match(reg);
+		if (!match) continue;
+
+		switch (mappingType) {
+			case MIDIMetaMappingType.ChannelPressure: {
+				const chanpress = parseInt(match.groups.chanpress, 10);
+				if (isNaN(chanpress) || chanpress < 1 || chanpress > 16) return false;
+				return {
+					type: MIDIMetaMappingType.ChannelPressure,
+					mapping: { chanpress } as MIDIChannelPressureMetaMapping
+				};
+			}
+			case MIDIMetaMappingType.ControlChange: {
+				const chan = parseInt(match.groups.chan, 10);
+				if (isNaN(chan) || chan < 1 || chan > 16) return false;
+
+				const ctrl = parseInt(match.groups.ctrl, 10);
+				if (isNaN(ctrl) || ctrl < 0 || ctrl > 127) return false;
```

Comment:
this `if` essentially happens a ton of times.. i'd probably make the whole thing into a func?

```
function parseMIDIByte(val: string, min: number, max: number) {
  let n = parseInt(val, 10);
  if (isNaN(n) || n < min || n > max) {
    return null;
  }
  return n;
}

const ctrl = parseMIDIByte(match.groups.ctrl, 0, 127);
if (ctrl == null) return false;